pkg/hydrate/controller.go

```diff
@@ -115,13 +115,28 @@ func (h *Hydrator) runHydrate(sourceCommit, syncDir string) HydrationError {
 	if err := kustomizeBuild(syncDir, dest, true); err != nil {
 		return err
 	}
+	if changed, commit := h.checkCommitChanged(sourceCommit); changed {
+		return NewInternalError(fmt.Errorf("source commit changed while running Kustomize build, was %s, now %s. It will be retried in the next sync", commit, sourceCommit))
```

Comment:
You need to swap the order of commit and sourceCommit. `commit` represents the new commit, while sourceCommit represents the old commit.

Reply:
Done